clarity/src/vm/docs/mod.rs

```diff
@@ -2189,6 +2189,28 @@ to deserialize the type, the method returns `none`.
 "#,
 };
 
+const REPLACE_AT: SpecialAPI = SpecialAPI {
+    input_type: "sequence_A, uint, A",
+    output_type: "(response sequence_A uint)",
+    snippet: "replace-at ${1:sequence} ${2:index} ${3:element}",
+    signature: "(replace-at sequence index element)",
+    description: "The `replace-at` function takes in a sequence, an index, and an element, 
+and returns a new sequence with the data at the index position replaced with the given element. 
+The given element's type must match the type of the sequence, and must correspond to a single 
+index of the input sequence. The return type on success is the same type as the input sequence. 
+
+If the provided index is out of bounds, this functions returns `(err u1)`.
```

Comment:
If there's no other way for this function to fail, then would it make sense to instead have `replace-at` return `(optional sequence_A)`?

Reply:
hmmm I think returning an error makes more sense - @kantai do you have an opinion?

Reply:
I don't have a strong opinion either way -- there's good arguments for either option:

In favor of `err`: public methods usually already use `uint` error types, so this lets them just use `try!` if they're kind of lazy.

In favor of `optional`: because there's no real error variation, `optional` matches the true type of the result more closely.